ansible/roles/jupyterhub/files/jupyterhub_config_lti11.py

```diff
@@ -214,7 +214,7 @@
 # this url pattern was changed to accept spaces in the assignment name
 c.JupyterHub.extra_handlers = [
     (
-        r'/submit-grades/(?P<course_id>\w+)/(?P<assignment_name>.*)$',
+        r'/submit-grades/(?P<course_id>[a-zA-Z0-9-_]+.*)/(?P<assignment_name>.*)$',
```

Comment:
It seems that regex have a conflict because with `.*` at the end we are accepting almost any character and the ranges a-zA-Z0-9 are ignored. Why not only use `(?P<course_id>[a-zA-Z0-9-_]+` and test it?

Reply:
You are correct. It doesn't make sense to accept any character after checking for the other characters.